src/nodeDebugAdapter.ts

```diff
@@ -636,13 +637,13 @@ export class NodeDebugAdapter extends ChromeDebugAdapter {
         }
     }
 
-    private async resolveProgramPath(programPath: string, sourceMaps: boolean): Promise<string> {
+    private async resolveProgramPath(programPath: string, sourceMaps: boolean, filePatterns: string[]): Promise<string> {
         logger.verbose(`Launch: Resolving programPath: ${programPath}`);
         if (!programPath) {
             return programPath;
         }
 
-        if (utils.isJavaScript(programPath)) {
```

Comment:
Is this the only usage of isJavascript? Can probably delete it.

Reply:
I ended up moving js determining to its own class that all usages of `isJavaScript` can share. We updated the stored patterns whenever we get a launch request. This seems to work well, I'm reasonably happy with this solution.